src/golang/jobs/test-nodb.yml

```diff
@@ -45,11 +53,65 @@ steps:
       condition: << parameters.checkout >>
       steps:
         - checkout
+  - when:
+      condition: << parameters.cache >>
+      steps:
+        - restore_cache:
+            keys:
+              - v1-mod-cache-{{ checksum "go.sum" }}
+              - v1-mod-cache-
+  - when:
+      condition: << parameters.check-modules >>
+      steps:
+        - go:
+            pkg-spec: ""
+            opts: mod verify
+        - go:
+            pkg-spec: ""
+            opts: mod tidy
+        - run:
+            name: Check Modules
+            command: |
+              if [ ! -z "$(git status --porcelain)" ]; then
+                echo
+                echo -e '\033[1;93m!!! "go mod verify && go mod tidy" resulted in changes. Please run those commands locally and commit the changes.\e[0m'
+                echo
+                git status
+                exit 1
+              fi
+  - when:
+      condition: << parameters.check-vendor >>
+      steps:
+        - run:
+            name: Check Vendor
+            command: |
+              if [ -d ./vendor ]; then
+                go mod vendor
+              else
+                echo
+                echo -e '\033[1;93m./vendor does not exist, but "check-vendor" is enabled. Disable check-vendor or populate a vendor folder\e[0m'
+                echo
+                exit 1
+              fi
+              if [ ! -z "$(git status --porcelain)" ]; then
+                git status
```

Comment:
I noticed that you run `git status` before the error message, but for `Check Modules` you run `git status` afterwards, was that intentional?